sample/src/main/kotlin/com/slack/circuit/sample/petdetail/PetDetail.kt

```diff
@@ -121,35 +124,49 @@ class PetDetailScreenFactory @Inject constructor() : ScreenViewFactory {
   }
 }
 
-private fun petDetailUi() = ui<PetDetailScreen.State, Nothing> { state, _ -> RenderImpl(state) }
+private fun petDetailUi() = ui<PetDetailScreen.State, Nothing> { state, _ -> PetDetail(state) }
 
 @Composable
-private fun RenderImpl(state: PetDetailScreen.State) {
+internal fun PetDetail(state: PetDetailScreen.State) {
   Scaffold(modifier = Modifier.systemBarsPadding()) { padding ->
     when (state) {
       PetDetailScreen.State.Loading -> {
         Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
-          CircularProgressIndicator()
+          CircularProgressIndicator(modifier = Modifier.testTag("progress"))
+        }
+      }
+      PetDetailScreen.State.UnknownAnimal -> {
+        Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+          Text(
+            modifier = Modifier.testTag("unknown animal"),
+            text = stringResource(id = R.string.unknown_animals)
+          )
         }
       }
-      PetDetailScreen.State.NoAnimal -> TODO()
       is PetDetailScreen.State.Success -> {
-        LazyColumn(modifier = Modifier.padding(padding)) {
+        LazyColumn(modifier = Modifier.padding(padding).testTag("animal container")) {
           item {
             AsyncImage(
-              modifier = Modifier.fillMaxWidth(),
+              modifier = Modifier.fillMaxWidth().testTag("image"),
               model =
                 ImageRequest.Builder(LocalContext.current)
                   .data(state.photoUrl)
+                  .fallback(R.drawable.dog)
```

Comment:
I'm not suuuuuuper stoked about this approach. Could we show some sort of error state instead?

Reply:
What don't you like? My thinking here is that we are assuming there will always be an image, which I don't think is a safe assumption. `fallback` just uses a default image if/when `state.photoUrl` is null.

I used that amazing picture of Baxter as it is what I had on hand. A inoffensive, graphic (similar to what Slack shows when a user doesn't have an avatar) would be better

Reply:
I don't have a good answer handy, but mostly it boils down to us checking in a test asset just to make the test happy. In production, folks will likely have other fallback options _or_ not want to always have to remember to set one. Maybe a good question to look/ask in the Coil repo for guidance?